sold.cc

```diff
@@ -384,8 +384,8 @@ uintptr_t Sold::TLSMemSize() const {
 void Sold::DecideMemOffset() {
     uintptr_t offset = 0x10000000;
     for (ELFBinary* bin : link_binaries_) {
-        const Range range = bin->GetRange() + offset;
-        CHECK(range.start == offset);
+        const Range range = bin->GetRange() + offset - bin->GetRange().start;
```

Comment:
Could you tell me which .so file has `bin->GetRange().start != 0`? I'm not sure if subtracting `start` is the right fix for it.  Have you considered a fix like

```
-        CHECK(range.start == offset);
-        offsets_.emplace(bin, range.start);
+        offsets_.emplace(bin, offset);
```

I'm not sure at all this is better. However, this makes a bit more sense to me than subtracting `start`.

Reply:
`bin->GetRange().start == 0` is true for executables.

Reply:
As I understood the intention of the code, I restore it and added a comment. https://github.com/shinh/sold/pull/75/commits/52807b61d5cfe99c990380b4e48e56a29cb9a887